File: 01_Language/01_Functions/php/next.php
<?php

$transport = array('foot', 'bike', 'car', 'plane');
$mode = current($transport); // $mode = 'foot';
echo $mode, "\n";
$mode = next($transport);    // $mode = 'bike';
echo $mode, "\n";
$mode = next($transport);    // $mode = 'car';
echo $mode, "\n";
$mode = prev($transport);    // $mode = 'bike';
echo $mode, "\n";
$mode = end($transport);     // $mode = 'plane';
echo $mode, "\n";

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/Next.java
package com.onevgo.functions;

import java.util.Arrays;
import java.util.Iterator;
import java.util.List;

public class Next {
    public static <T> T next(Iterator<T> array) {
        return array.next();
    }

    public static void main(String[] args) {
        List<String> transport = Arrays.asList("foot", "bike", "car", "plane");
        Iterator<String> iterator = transport.listIterator();
        System.out.println(next(iterator));
        System.out.println(next(iterator));
        System.out.println(next(iterator));
        System.out.println(next(iterator));
    }
}
